ocs_ci/ocs/benchmark_operator.py

```diff
@@ -0,0 +1,199 @@
+            run_cmd('oc apply -f my_custom_bench')
+
+        """
+        log.info("Initialize the benchmark-operator object")
+        self.args = kwargs
+        self.repo = self.args.get("repo", BMO_REPO)
+        self.branch = self.args.get("branch", "master")
+        # the namespace is a constant for the benchmark-operator
+        self.namespace = BMO_NAME
+        self.pgsql_is_setup = False
+        self.ocp = OCP()
+        self.ns_obj = OCP(kind="namespace")
+        self.pod_obj = OCP(namespace=BMO_NAME, kind="pod")
+        # list of worker nodes to label
+        self.worker_nodes = [node.name for node in get_nodes()]
+        self._clone_operator()
+        self.dir += f"/{BMO_NAME}"
+
+        # to use the cache dropping pod, worker nodes need to be labeled.
+        log.info("Labeling the worker nodes for cache-dropping enable.")
+        try:
+            helpers.label_worker_node(
+                self.worker_nodes, label_key=BMO_LABEL, label_value="yes"
+            )
+        except CommandFailed:
+            # this is probably because of the nodes are already labeled, so,
+            # checking if nodes are labeled and continue anyway.
+            result = self.pod_obj.exec_oc_cmd(f"get node -l {BMO_LABEL}")
+            found = [node for node in self.worker_nodes if re.search(node, result)]
+            if len(found) == len(self.worker_nodes):
+                log.info("All worker nodes are labeled")
+            else:
+                log.warning("Labeling nodes failed, Not all workers node are labeled !")
+
+    def _clone_operator(self):
+        """
+        clone the benchmark-operator repo into temp directory
+
+        """
+        self.dir = tempfile.mkdtemp(prefix=f"{BMO_NAME}_")
+        try:
+            log.info(f"Cloning {BMO_NAME} in {self.dir}")
+            git_clone_cmd = f"git clone -b {self.branch} {self.repo} --depth 1"
+            run(git_clone_cmd, shell=True, cwd=self.dir, check=True)
+        except (CommandFailed, CalledProcessError) as cf:
+            log.error(f"Error during cloning of {BMO_NAME} repository")
+            raise cf
+
+    def deploy(self):
+        """
+        Deploy the benchmark-operator
+
+        """
+        log.info("Deploy the benchmark-operator project")
+        try:
+            run("make deploy", shell=True, check=True, cwd=self.dir)
+        except Exception as ex:
+            log.error(f"Failed to deploy benchmark operator : {ex}")
+
+        log.info("Wait for the benchmark-operator deployment be available")
+        try:
+            cmd = f'wait --for=condition=available "{BMO_DEPLOYMENT}" -n {BMO_NAME} --timeout=300s'
+            self.pod_obj.exec_oc_cmd(cmd)
+        except Exception as ex:
+            log.error(f"Failed to wait for benchmark operator : {ex}")
+
+        # since the Benchmark-Operator pod have 2 containers it get into Running
+        # state and not both of containers are up, so, wait 5 sec until 2/2 are up
+        time.sleep(5)
```

Comment:
@Avilir I would suggest to add respective pod validation here to be up and running instead of sleep, we might see delay in pod running state wrt different platforms

Reply:
changed, and now i am testing for containers status and not just sleeping.

thanks.